src/graph.js

```diff
@@ -0,0 +1,94 @@
+'use strict'
+
+import deepEqual from 'deep-equal'
+import _ from 'lodash'
+
+class Graph {
+  constructor () {
+    this._clients = {}
+    this._connections = []
+  }
+
+  addClient (client) {
+    this._clients[client.name] = this._clients[client.name] || client
```

Comment:
no.
Scenario:
1/ User runs a client with input inMedia and output outMedia
2/ User adds an input
He stops the program and re-run it.